pkg/metrics/exporter.go

```diff
@@ -48,20 +53,55 @@ func RegisterPrometheusExporter(registry *prom.Registry) (http.Handler, error) {
 
 // RegisterStackdriverExporter register a Stackdriver exporter to OpenCensus.
 // It will add Agones metrics into Stackdriver on Google Cloud.
-func RegisterStackdriverExporter(projectID string) (sd *stackdriver.Exporter, err error) {
-	// Default project will be used
-	sd, err = stackdriver.NewExporter(stackdriver.Options{
+func RegisterStackdriverExporter(projectID string, defaultLabels string) (*stackdriver.Exporter, error) {
+	instanceID, err := metadata.InstanceID()
+	if err != nil {
+		logger.WithError(err).Warn("error getting instance ID")
+		instanceID = unknown
+	}
+	zone, err := metadata.Zone()
+	if err != nil {
+		logger.WithError(err).Warn("error getting zone")
+		zone = unknown
+	}
+	clusterName, err := metadata.InstanceAttributeValue("cluster-name")
+	if err != nil {
+		logger.WithError(err).Warn("error getting cluster-name")
+		clusterName = unknown
+	}
+
+	labels, err := parseLabels(defaultLabels)
+	if err != nil {
+		return nil, err
+	}
+
+	sd, err := stackdriver.NewExporter(stackdriver.Options{
 		ProjectID: projectID,
 		// MetricPrefix helps uniquely identify your metrics.
 		MetricPrefix: "agones",
+		Resource: &monitoredres.MonitoredResource{
+			Type: "k8s_container",
+			Labels: map[string]string{
+				"project_id":   projectID,
+				"instance_id":  instanceID,
```

Comment:
Do these have to be set? Or can they be left blank if we can't read from the metadata server?